Mathlib/FieldTheory/PurelyInseparable.lean

```diff
@@ -537,6 +537,32 @@ theorem IsPurelyInseparable.injective_comp_algebraMap [IsPurelyInseparable F E]
   haveI := expChar_of_injective_ringHom (f.comp (algebraMap F E)).injective q
   exact iterateFrobenius_inj L q n heq
 
+theorem IsPurelyInseparable.injective_comp_toAlgHom [IsPurelyInseparable F E] (R L) [CommSemiring R]
+    [CommRing L] [IsReduced L] [Algebra R F] [Algebra R E] [Algebra R L] [IsScalarTower R F E] :
+    Function.Injective fun f : E →ₐ[R] L ↦ f.comp (IsScalarTower.toAlgHom R F E) := fun _ _ eq ↦
```

Comment:
I think maybe `fun f : E →ₐ[R] L ↦ f.comp (IsScalarTower.toAlgHom R F E)` is almost the same as `AlgHom.restrictDomain`.

Reply:
I think now the order of variables in the definition of `AlgHom.restrictDomain` is not well arranged, which causes a subtle difference between it and `fun f : E →ₐ[R] L ↦ f.comp (IsScalarTower.toAlgHom R F E)`. Therefore, maybe order of variables in the definition of `AlgHom.restrictDomain` should be changed, so that we could directly write `AlgHom.restrictDomain` here.